### PR TITLE
Register str_extract as Macro

### PR DESCRIPTION
**Pull request by CupOfTea696**
Move the `str_extract` code to the String Macros.

### CODE REVIEW

File: src/StrMacros.php [hunk 1]
```diff
@@ -14,10 +14,24 @@ public function between()
...
             }
+
             return Str::before(Str::after($subject, $beginning), $end);
         };
     }
 
+    public function extract()
+    {
+        return function ($string, $pattern) {
+            if (@preg_match($pattern, $string) === false) {
```
Comment (peter279k) [inline src/StrMacros.php]: It looks like the `preg_match` will return integer value.

And I think this will not return the bool value.

Comment (CupOfTea696) [reply src/StrMacros.php]: From [php.net](https://www.php.net/manual/en/function.preg-match.php#refsect1-function.preg-match-returnvalues):

"**preg_match()** returns 1 if the **pattern** matches given **subject**, 0 if it does not, or **FALSE** if an error occurred.

Comment (peter279k) [reply src/StrMacros.php]: That makes sense.

**Comment by calebporzio** [comment]
Thanks @CupOfTea696 !

**Pull request merged**